packages/lisk-p2p/src/peer.ts

```diff
@@ -316,6 +316,13 @@ export class Peer extends EventEmitter {
 		this._outboundSocket.connect();
 	}
 
+	public createPeerFromOutboundConnection(
```

Comment:
We should pass the outbound socket to the peer's constructor as an optional argument instead of having a separate `createPeerFromOutboundConnection` method. This is dangerous here because it could be called multiple times. If we move this logic to the constructor then we guarantee that this cannot happen.

The logic which instantiates the `Peer` can just set the `inboundSocket` as `undefined` explicitly if there is no inbound socket.